emv2/org.osate.xtext.aadl2.errormodel/src/org/osate/xtext/aadl2/errormodel/validation/ErrorModelJavaValidator.java

```diff
@@ -444,6 +445,21 @@ private void checkSConditionElementType(SConditionElement conditionElement) {
 		}
 	}
 
+	private void checkConditionElementDirection(ConditionElement conditionElement) {
+		ErrorPropagation ep = EMV2Util.getErrorPropagation(conditionElement.getQualifiedErrorPropagationReference());
+		Subcomponent sub = EMV2Util.getLastSubcomponent(conditionElement.getQualifiedErrorPropagationReference());
+		if (ep == null) {
+			return;
+		}
+		if (sub != null && ep.getDirection() != DirectionType.OUT) {
+			error(conditionElement,
+					"Reference to subcomponent propagation " + EMV2Util.getPrintName(ep) + " must be out.");
+		} else if (ep.getDirection() != DirectionType.IN) {
+			error(conditionElement,
+					"Reference to subcomponent propagation " + EMV2Util.getPrintName(ep) + " must be in.");
+		}
```

Comment:
On the original model I get errors "Reference to subcomponent propagation XXX must be in." However, the reference is not to a subcomponent propagation, so the error message is wrong.